Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -317,53 +323,73 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
 
     // MARK: Channel handler actions
     
-    private func run(_ action: ConnectionStateMachine.ConnectionAction, with context: ChannelHandlerContext) {
+    private func run(
+        _ action: ConnectionStateMachine.ConnectionAction,
+        with context: ChannelHandlerContext,
+        promise runPromise: EventLoopPromise<Void>?
```

Comment:
this should not be part an extra argument here. Instead we should change: `ConnectionStateMachine.ConnectionAction`'s cases:

```
        case sendParseDescribeBindExecuteSync(PostgresQuery)
        case sendBindExecuteSync(PSQLExecuteStatement)
```

to also contain an optional writePromise.